api/authorization-api-1_0.md

```diff
@@ -328,77 +329,256 @@ The following is a non-normative example of a simple Decision:
+- `reason`
+- `properties`
+- `obligations`
+- `metadata`
+- `environment`
 
-### Example Context
-An implementation MAY follow a structured approach to `"context"`, in which it presents the reasons that an authorization request failed.
+Any of these context fields MAY contain natual language messages or explanations. Internationalization (i18n) is a Client and User Experience (UX) concern and out of scope for this specification. Implementations of this specification MAY choose to return multi-lingual messages, but in that case the PEP SHOULD also be able to process these internationalized responses.
 
-- A list of identifiers representing the items (policies, graph nodes, tuples) that were used in the decision-making process.
-- A list of reasons as to why access is permitted or denied.
+These context response fields are detailed in the following sections...
 
-#### Reasons
-Reasons MAY be provided by the PDP. 
+#### Reason context field
+OPTIONAL - The `reason` element is a JSON Array object representing the Reasons why the decision was made, typically in natural language or some other implementation-specific human-readable format. This object can convey information that can be used by the PEP as part of the access enforcement process, or just passed back to the requesting client. The Reason object clarifies and complements the decision, whether access was granted or not.
+
+A PEP can ignore this field if it doesn't know how to process it.
+
+The Reason object being a JSON object, implementations MUST follow the following structure; the `reason` context contains the following elements:
+
+`id`:
+: REQUIRED. A string value that specifies the reason within the scope of a particular response. In case the PDP returns several reasons, the `id` MUST uniquely identify each one of them. The `id` MAY also match specific reasons held by the PDP, part of a reasons list or dictionnary for example, and used for audit correlations.
+
+`reason_admin`:
+: OPTIONAL. The reason, which MUST NOT be shared with the end user, but useful for administrative purposes that indicates why the access was granted or denied. The value of this field is a Reason Field object ({{reason-field}}).
+
+`reason_user`:
+: OPTIONAL. The reason, which MAY be shared with the end user that indicates why the access was granted or denied. The value of this field is a Reason Field object ({{reason-field}}). 
 
 ##### Reason Field {#reason-field}
-A Reason Field is a JSON object that has keys and values of type `string`. The following are non-normative examples of Reason Field objects:
+A Reason Field is a JSON object with `String` key and value pairs, which has the following form:
+
+`{ "reason-code": "reason-description"}`
+
+where:
+
+`reason-code`:
+: REQUIRED. A String value representing a code that uniquely identifies the error. Implementations that do not use any error-code mechanism SHOULD default to valid HTTP Error codes (see {{RFC7231}} ).
```

Comment:
How does the reason-code relate to the `id` in the reason object?  It seems like the `id` is some arbitrary number, whereas the `reason-code` is meant to be an HTTP error code. Is there a reason why we need both?